lib/ex_aws/operation/s3.ex

```diff
@@ -63,6 +64,19 @@ defmodule ExAws.Operation.S3 do
       {operation |> Map.put(:path, path), config}
     end
 
+    @spec normalize_path(operation :: ExAws.Operation.S3.t()) ::
+            ExAws.Operation.S3.t()
+    def normalize_path(operation) do
+      normalized_path =
+        if String.first(operation.path) === "/" do
+          operation.path
+        else
+          Path.join(["/", operation.path])
+        end
+
+      operation |> Map.put(:path, normalized_path)
+    end
+
```

Comment:
Could this whole thing perhaps be simplified to something like:

```elixir
defp ensure_absolute_path(operation), do: put_in(operation.path, Path.join("/", operation.path])
```

since `Path.join/2` won't add supurflous slashes anyway?

Reply:
Thanks for the idea, I changed it